src/main/java/uk/gov/justice/digital/hmpps/whereabouts/services/AppointmentService.kt

```diff
@@ -101,29 +110,63 @@ class AppointmentService(
 
       recurringAppointmentRepository.save(recurringAppointment)
 
-      raiseRecurringAppointmentTrackingEvent(createAppointmentSpecification, createAppointmentSpecification.repeat)
+      raiseRecurringAppointmentCreatedTrackingEvent(
+        createAppointmentSpecification,
+        createAppointmentSpecification.repeat
+      )
     }
 
     return appointmentCreated
   }
 
+  @Transactional
+  fun deleteAppointment(appointmentId: Long) {
+    prisonApiService.getPrisonAppointment(appointmentId)
+      ?: throw EntityNotFoundException("Appointment $appointmentId does not exist")
+
+    val videoLinkBooking: VideoLinkBooking? =
+      videoLinkBookingRepository.findByMainAppointmentIds(listOf(appointmentId)).firstOrNull()
+
+    if (videoLinkBooking != null) {
```

Comment:
Just checking - we can't have recurring VLBs?

Reply:
We can't at the minute. The design team are confident that we won't in the future.